src/Interactive/EditorFeatures/Core/Extensibility/Interactive/InteractiveEvaluator.cs

```diff
@@ -438,7 +438,22 @@ public virtual bool CanExecuteCode(string text)
 
         public Task<ExecutionResult> ResetAsync(bool initialize = true)
         {
-            GetInteractiveWindow().AddInput(_interactiveCommands.CommandPrefix + "reset");
+            // This is an implementation of ResetAsync on IInteractiveEvaluator interface.
+            // C# and VB Interactive window should not be calling this method at all
+            // as they now call reset through IInteractiveEvaluator2
+
+            // Now we pass on this call to IInteractiveEvaluation2 implementation
+            // which maintains old behaviour when isFromPrompt was not there
+            
+            return ResetAsync(initialize, true);
+        }
+
+        public Task<ExecutionResult> ResetAsync(bool initialize = true, bool isFromPrompt = true)
+        {
+            if (!isFromPrompt)
+            {
+                GetInteractiveWindow().AddInput(_interactiveCommands.CommandPrefix + "reset");
```

Comment:
For commands like reset and cls, we need slightly different handling based on where they originated ( at prompt or through UI) .Displaying the command at the prompt is one such case. There is also logic to save uncommitted inputs which needs to be handled when these commands are coming from UI but the scenario does not make sense if they are coming from prompt ( as there would be no uncommitted input in this case). Uncommitted input is used to display the text user had typed at the prompt ( may be a long multi line submission) before he hits reset or cls.